cachito/web/models.py

```diff
@@ -231,6 +232,26 @@ def to_purl(self):
                     f"Unknown protocol in npm package version: {self.version}"
                 )
 
+        elif self.type == "pip":
+            # As per the purl spec, PyPI names should be normalized by lowercasing and
+            # converting '_' to '-'. The safe_name() function does the latter but not the
+            # former. It is not necessary to escape characters in the name, safe_name()
+            # also replaces everything except alphanumeric chars and '.' with '-'.
+            name = pkg_resources.safe_name(self.name.lower())
+            version = self.version
+
+            if version.startswith("vcs:"):
+                quoted_url = urllib.parse.quote(version[len("vcs:") :], safe="")
+                return f"pkg:generic/{name}?vcs_url={quoted_url}"
```

Comment:
In the design doc, the format for Git dependencies is defined as `pkg:/github/...`.

However, for npm, we use `pkg:generic` with `vcs_url` if the dependency uses `git+<protocol>://`, see here: https://github.com/release-engineering/cachito/blob/28df0cfc3b6c0d5f09a1792fc3fa82ca7945d9d2/cachito/web/models.py#L223-L225

For Pip, all Git dependencies will use `git+<protocol>://`, so I figured I would keep this consistent with npm, WDYT?